js-miniapp-sample/src/pages/auth-token.js

```diff
@@ -65,53 +78,86 @@ const dataFetchReducer = (state, action) => {
         ...state,
         isLoading: true,
         isError: false,
-        response: null,
+        hasRequestedPermissions: false,
       };
     case 'FETCH_SUCCESS':
       return {
         ...state,
         isLoading: false,
         isError: false,
-        response: action.tokenData,
+        hasRequestedPermissions: true,
       };
     case 'FETCH_FAILURE':
       return {
         ...state,
         isLoading: false,
         isError: true,
-        errorMessage: action.errorMessage,
       };
     default:
       throw new Error();
   }
 };
 
-function AuthToken() {
+type AuthTokenProps = {
+  permissions: CustomPermissionName[],
+  accessToken: AccessTokenData,
+  getAccessToken: () => Promise<string>,
+  requestPermissions: (
+    permissions: CustomPermission[]
+  ) => Promise<CustomPermissionResult[]>,
+};
+
+function AuthToken(props: AuthTokenProps) {
   const [state, dispatch] = useReducer(dataFetchReducer, initialState);
   const classes = useStyles();
 
   const buttonClassname = clsx({
     [classes.buttonFailure]: state.isError,
-    [classes.buttonSuccess]: state.response,
+    [classes.buttonSuccess]: !state.isError,
   });
 
-  function requestToken() {
-    MiniApp.user
-      .getAccessToken()
-      .then((response) => {
-        dispatch({ type: 'FETCH_SUCCESS', tokenData: response });
-      })
-      .catch((error) => {
-        console.error(error);
-        dispatch({ type: 'FETCH_FAILURE', errorMessage: error });
+  function requestAccessTokenPermission() {
+    const permissionsList = [
+      {
+        name: CustomPermissionName.ACCESS_TOKEN,
+        description:
+          'We would like to get the Access token details to share with this Mini app',
+      },
+    ];
+
+    props
+      .requestPermissions(permissionsList)
+      .then((permissions) =>
+        permissions
+          .filter(
+            (permission) => permission.status === CustomPermissionStatus.ALLOWED
+          )
+          .map((permission) => permission.name)
+      )
+      .then((permissions) =>
+        Promise.all([
+          hasPermission(CustomPermissionName.ACCESS_TOKEN, permissions)
+            ? props.getAccessToken()
+            : null,
+        ])
+      )
+      .then(() => dispatch({ type: 'FETCH_SUCCESS' }))
+      .catch((e) => {
+        console.error(e);
+        dispatch({ type: 'FETCH_FAILURE' });
       });
   }
 
+  function hasPermission(permission, permissionList: ?(string[])) {
+    permissionList = permissionList || props.permissions || [];
+    return permissionList.indexOf(permission) > -1;
+  }
+
   function handleClick(e) {
     if (!state.isLoading) {
       e.preventDefault();
       dispatch({ type: 'FETCH_INIT' });
-      requestToken();
+      requestAccessTokenPermission();
```

Comment:
When I test on iOS simulator, cannot get token or permission request.
[Screenshot](https://user-images.githubusercontent.com/60589925/109097611-00927d00-7763-11eb-8447-f25311ecb020.png)

Reply:
As discussed, let's verify this with local build bridge.js.

Reply:
I guess you need to check the build from `feature` branch, not master, as we haven't merged the access token changes in iOS master branch yet. However, I see there is an issue with the flow, working on it.

Reply:
Done